dashboard/app/views/followers/_progress3.html.haml

```diff
@@ -1,296 +0,0 @@
-              .stage{style: "width: #{width * widthpx}px;"}
-                &nbsp;    
-          .studentbar{style: "position: absolute; display: none", id: "studentbar_#{student.id}"}
-          .studentdot{style: "position: absolute; display: none, left: -15px; padding-top:4px; padding-bottom:4px", id: "studentdot_#{student.id}"}
-            .dot
-              35
-   
-        .clear
-
-          
-    :javascript
-      var studentIds = JSON.parse('#{student_ids.to_json}'); 
-      var allLevels = JSON.parse('#{all_levels.to_json}');
-      var stageLengths = JSON.parse('#{stage_lengths}');
-%br/
-
-.key
-  %dl
-    %dt
-      %span.bar.attempted &nbsp;
-    %dd= t('progress.not_completed')
-    %dt
-      %span.bar.passed &nbsp;
-    %dd= t('progress.completed_too_many_blocks')
-    %dt
-      %span.bar.perfect &nbsp;
-    %dd= t('progress.completed_perfect')
-    %dt
-      .dot 10
-    %dd= t('progress.furthest_level_attempted')
-
-:javascript
-  function progressLoaded() {
-    // Rendering with javascript here gives us the option of doing a version that updates in realtime with ajax calls.
-
-    for (var s = 0; s < studentIds.length; s++) {
-      var highestLevel = 0;
-      var levels = allLevels[s];
-      var studentId = studentIds[s];
-
-      // find the furthest level that this student has attempted
-      for (var l = levels.length - 1; l > 0; l--) {
-        if (levels[l]) {
-          highestLevel = l;
-          break;
-        }
-      }   
-
-      var classForLevelInfo = function(levelInfo) {
-        if (levelInfo) {
-          if (levelInfo.best_result >= #{Activity::FREE_PLAY_RESULT}) {
-            return  'perfect'; // perfect (or free play)
-          } else if (levelInfo.best_result >= #{Activity::MINIMUM_PASS_RESULT}) {
-            return 'passed'; // passed but not perfect
-          } else {
-            return 'attempted'; // attempted, not passed
-          }
-        }
-        return '';
-      };
-
-      var stringForLevelInfo = function(levelInfo) {
-        if (levelInfo) {
-          if (levelInfo.best_result >= #{Activity::FREE_PLAY_RESULT}) {
-            return "#{t('progress.completed_perfect')}";
-          } else if (levelInfo.best_result >= #{Activity::MINIMUM_PASS_RESULT}) {
-            return '#{t("progress.completed_too_many_blocks")}';
-          } else {
-            return '#{t("progress.not_completed")}';
-          }
-        }
-        return '';
-      };
-
-
-      /* Draw the level history, though it'll be hidden until we fade it in. */
-      for (var l = 0; l <= highestLevel; l++)
-      {
-        var levelInfo = allLevels[s][l];
-
-        var id = "#studentbar_" + studentId;
-        var title = "Level " + (getLevelNumberInStage(l) + 1) + " " + stringForLevelInfo(levelInfo);
-        $(id).append($('<div>&nbsp;</div>').addClass('bar').addClass(classForLevelInfo(levelInfo)).attr('title', title));
-      }
-
-      /* Animate each student's dot to the correct level. */
```

Comment:
Hadi really liked the animation when he saw it :)  Also, it was designed so that we could add realtime updating and animate realtime student progress.